#123 Implement display of row_number parameter for child tasks

cnstlungu opened the pull request:
Hello everyone,

This is my proposal for the changes. Please peer review and suggest any changes or improvements. 

It is my understanding that only RowEntry tasks need a row_number

Short note on how testing was done:
1. Get a parent task and provide it a number
2. Verify a child task with the given number is generated and displayed
3. Fill in child task
4. Verify data has been inserted in the database

**Testing PASSED**

TaskOwnedLandRowEntry OK
TaskOwnedAutomobileRowEntry OK
TaskOwnedBankAccountsRowEntry OK
TaskOwnedDebtsRowEntry OK
TaskOwnedIncomeFromPensionsRowEntry OK
TaskOwnedJewelryRowEntry OK
TaskExtraValuableRowEntry OK
TaskOwnedBuildingsRowEntry OK
TaskOwnedIncomeFromInvestmentsRowEntry OK

**Testing FAILED due to existing problems:**

- _TaskTranscribeOwnedInvestmentsTable_ => 'NoneType' object has no attribute 'row_number' => happens because there is no child_class => Is this (and its corresponding model) a duplicate that is not needed? We already have _TaskOwnedIncomeFromInvestmentsRowEntry_.
- _TaskOwnedIncomeFromIndependentActivitiesTable_ => 'NoneType' object has no attribute 'row_number' => because there is no child_class
- _TaskOwnedIncomeFromAgriculturalActivitiesRowEntry_ =>  Cannot resolve keyword 'annual_income_currency' into field. => field vs model mismatch => bug exists (#119)
- _TaskOwnedGoodsOrServicesRowEntry_ => Cannot resolve keyword 'person' into field. 
- TaskOwnedIncomeFromDeferredUseOfGoodsRowEntry => Cannot resolve keyword 'person' into field.
- TaskOwnedIncomeFromGamblingRowEntry => Cannot resolve keyword 'person' into field.
- TaskOwnedIncomeFromOtherSourcesRowEntry => Cannot resolve keyword 'address' into field. => wrong class used in save_verified_data method => bug exists ( #154  )
- TaskOwnedIncomeFromSalariesRowEntry => Cannot resolve keyword 'person' into field.


Kindly let me know if it's OK for me to create bugs for the tasks currently failing that aren't logged as such yet.

Regards
Costin

RaduCStefanescu commented:
hey @cnstlungu thank you for the pull request, we are reviewing it and probably will merge it soon. In the mean time, if you would be able to log defects for the issues that are failing the tests, that would be awesome.

cnstlungu commented:
@catileptic  @RaduCStefanescu  I've sorted out conflicts for this pull request. I've also retested all the tasks. All tasks work now - I can confirm I'm able to fill both RowEntry and Table data and see it pass through to the database together with the row_number. Please peer review.

A couple of more fixes were necessary and introduced:
- For TaskOwnedInvestmentsOver5K, I've changed get_or_create to use  'beneficiary_name'/'beneficiary_surname' instead of 'name'/'surname' as it was failing.
- For TaskOwnedIncomeFromOtherSources and TaskOwnedIncomeFromIndependent, I've added row_number (last time was unable to test them since they were failing).

Kindly let me know if there's anything else needed here to get this going forward.

Regards
Costin
